library/item/item_setting/src/main/java/org/quiltmc/qsl/item/setting/api/RecipeRemainderLocation.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2024 The Quilt Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.item.setting.api;
+
+import java.util.Objects;
+
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Contract;
+
+import net.minecraft.util.Identifier;
+
+import org.quiltmc.qsl.item.setting.impl.RecipeRemainderLogicHandlerImpl;
+
+/**
+ * Contains the different recipe remainder locations that QSL supports.
+ * Calling {@link #getOrCreate(Identifier)} allows mods to create their own remainder locations or get remainder locations without needing to compile against the other mod.
+ *
+ * <p> This class should not be extended.
+ */
+@ApiStatus.NonExtendable
+public interface RecipeRemainderLocation {
+	/**
+	 * Remainder location for Vanilla crafting. Used in Crafting Tables and the inventory crafting screen.
+	 */
+	RecipeRemainderLocation CRAFTING = getOrCreate(new Identifier("minecraft:crafting"));
```

Comment:
Just a nitpick, but `new Identifier("minecraft", "crafting")` or `new Identifier("crafting")` are clearer/neater IMO.

Reply:
The first is more characters, so no real improvement, and the latter hides `minecraft`, which i want to stay clear.